src/main/java/bio/terra/cda/app/util/NestedColumn.java

```diff
@@ -0,0 +1,64 @@
+package bio.terra.cda.app.util;
+
+public class NestedColumn {
+  private String column;
+  private String unnestClause;
+
+  public NestedColumn() {}
+
+  public NestedColumn(String column, String unnestClause) {
+    this.column = column;
+    this.unnestClause = unnestClause;
+  }
+
+  public static NestedColumn generate(String qualifiedColumnName) {
+    return Builder.generate(qualifiedColumnName);
+  }
+
+  public static class Builder {
+    /**
+     * column -> SELECT DISTINCT(column)
+     *
+     * <p>D.column -> SELECT DISTINCT(_D.column) FROM TABLE, UNNEST(D) AS _D
+     *
+     * <p>A.B.C.D.column -> > SELECT DISTINCT(_D.column) FROM TABLE, UNNEST(A) AS _A, UNNEST(_A.B)
+     * AS _B, UNNEST(_B.C) AS _C, UNNEST(_C.D) AS _D
+     */
+    public static NestedColumn generate(String qualifiedColumnName)
+        throws IllegalArgumentException {
+      StringBuilder unnestClause = new StringBuilder();
+      String newColumn = null;
+      if (qualifiedColumnName != null) {
+        String[] c = qualifiedColumnName.split("\\.");
+        if (c.length > 1) {
+          newColumn = "_" + c[c.length - 2] + "." + c[c.length - 1];
+          unnestClause.append(", UNNEST(" + c[0] + ") AS _" + c[0]);
+          for (int n = 1; n < c.length - 1; n++) {
+            unnestClause.append(", UNNEST(_" + c[n - 1] + "." + c[n] + ") AS _" + c[n]);
+          }
+          return new NestedColumn(newColumn, unnestClause.toString());
+        } else if (c.length == 1) {
+          return new NestedColumn(qualifiedColumnName, "");
+        }
```

Comment:
unnecessary `else` after `return`
```suggestion
        }
        if (c.length == 1) {
          return new NestedColumn(qualifiedColumnName, "");
        }
```

Reply:
Fixed